ntp-udp/src/lib.rs

```diff
@@ -1,225 +1,5 @@
 mod interface_name;
+mod raw_socket;
 mod socket;
 
 pub use socket::UdpSocket;
-use std::os::unix::prelude::{AsRawFd, RawFd};
```

Comment:
IMO splitting this in multiple tiny modules just makes it harder to understand how these pieces fit together. Also not clear if this commit is strictly moving code around or if it also changes the code in other ways?

Reply:
Added more text to the PR description explaining what this PR does.

As to the tiny module sizes, this was an explicit choice. Due to how visibility works in rust, all code in a module and its submodules must be considered when deciding whether an unsafe statement occuring in that module is safe. This makes it hard to reason about safety in larger modules.

Because of the potential impact of unsafety, I decided here to prioritize the ease of reasoning about safety over the clarity of the larger scale structure. I will add additional comments to the modules to explain the structure and the reasoning behind it.

Reply:
That's fair, but in that case I would advocate using inline modules (`mod foo { }`) instead of modules defined through a separate file.

Reply:
seconding this. I personally dislike having many files (makes navigating harder to me), and the only goal here is namespacing/isolation so inline modules seem like a good fit.

Reply:
I have done the flattening into a single file as a final commit on the end. If it is highly desirable for review to have this flattening applied throughout the history please let me know.

Reply:
Reordered to keep the original ordering in that file. There still is one ordering change, namely that the cerr function moved to the top, but hopefully that is not too distracting. Please let me know if this flattening is highly desirable to be done throughout the commit history for this PR.